src/custom-widget.ts

```diff
@@ -190,11 +187,53 @@ window.TagoIO.deleteData = (variables, callback): Promise<IData> | void => {
 
   // If a callback is not passed it returns the promise
   if (window.Promise && !callback) {
-    return new Promise((resolve: (data: IData) => void, reject: (data: IError) => void) => {
-      pool[uniqueKey] = (success: IData, error: IError): void => {
+    return new Promise((resolve: (data: TData) => void, reject: (data: TError) => void) => {
+      pool[uniqueKey] = (success: TData, error: TError): void => {
         if (error) reject(error);
         resolve(success);
       };
     });
   }
 };
+
+const editResourceData = (
+  variables: TDataRecord | TDataRecord[],
+  callback?: TSendDataCallback
+): Promise<TData> | void => {
+  const uniqueKey: string = shortid.generate();
+  pool[uniqueKey] = callback || null;
+  const variablesToEdit = Array.isArray(variables) ? variables : [variables];
+
+  sendMessage({
+    variables: variablesToEdit,
+    method: "edit-resource",
+    key: uniqueKey,
+  });
+
+  if (window.Promise && !callback) {
+    return new Promise((resolve: (data: TData) => void, reject: (data: TError) => void) => {
+      pool[uniqueKey] = (success: TData, error: TError): void => {
+        if (error) reject(error);
+        resolve(success);
+      };
+    });
+  }
+};
+
+const openLink: TTagoIO["openLink"] = (url) => {
```

Comment:
New function to open a link using the router navigation on Admin, making it possible for developers using this library to call `window.TagoIO.openLink("url")` and if the URL matches the pattern (e.g. `/dashboards/info`), Admin/RUN can use the router functions to browse without opening the URL inside of the Iframe.